src/bin/sol-fbp-runner/runner.c

```diff
@@ -73,18 +74,22 @@ search_fbp_file(struct runner *r, const char *basename)
     uint16_t i;
     int err;
 
-    SOL_PTR_VECTOR_FOREACH_IDX (r->fbp_paths, p, i) {
-        err = asprintf(&fullpath, "%s/%s", p, basename);
-        if (err < 0) {
-            errno = ENOMEM;
-            return NULL;
-        }
-
-        if (stat(fullpath, &s) == 0 && S_ISREG(s.st_mode))
-            return fullpath;
-
-        free(fullpath);
-    }
+#define STAT_PATH(_path) \
+    do { \
+        err = asprintf(&fullpath, "%s/%s", _path, basename); \
+        if (err < 0) { \
+            errno = ENOMEM;\
+            return NULL; \
+        } \
+        if (stat(fullpath, &s) == 0 && S_ISREG(s.st_mode)) \
+            return fullpath; \
+        free(fullpath); \
+    } while (0)
```

Comment:
Macros help in some cases, e.g. when you have a lot of shared context that would mean passing many arguments to a function. However they complicate in others, e.g. making it difficult to debug, giving less room for compiler to make decisions about inlining or not.

I prefer we stay with functions as a default form abstraction and use the macros when there's a need for them (since we pay a cost).

Sorry if my first message sounded harsh, the fix itself is good!